src/app.py

```diff
@@ -335,11 +335,28 @@ def query(
                 rule_author=rule.author,
                 is_global=rule.is_global,
                 is_private=rule.is_private,
+                is_degenerate=rule.parse().is_degenerate,
```

Comment:
We could probably call `rule.parse()` just once, somewhere above? In case it gets expensive in the future

Reply:
Performance hit is negliglible, and even if it wasn't, it's already cached:

```python
    def parse(self) -> UrsaExpression:
        if self.__parsed is None:
            self.__parsed = self.__parse_internal()
        return self.__parsed
```

I worry more about readability. There's no way to assign it to a temporary variable in a list comprehension, but I can rewrite this all to an explicit for loop if you prefer :thinking: It would look like this:

```python
response_rules = []
for rule in rules:
    parsed_rule = rule.parse()
    response_rules.append(
        ParseResponseSchema(
            rule_name=rule.name,
            rule_author=rule.author,
            is_global=rule.is_global,
            is_private=rule.is_private,
            is_degenerate=parsed_rule.is_degenerate,
            parsed=parsed_rule.query,
        )
    )
return response_rules
```

I prefer the current version, but I'm not overly attached to it if you prefer this one.